### PR TITLE
Issue 124

### PR DESCRIPTION
**Pull request by BasThomas**

### CODE REVIEW

**Comment by BasThomas** [comment]
@fassko opening this for transparency and to make it easier to discuss some details :)

File: _drafts/2018-12-13-issue-124.md [hunk 1]
```diff
@@ -21,27 +25,49 @@ sponsor:
...
+
+Amazon has open sourced [`smoke-framework`](https://github.com/amzn/smoke-framework), a lightweight server-side service framework written in the Swift and using [SwiftNIO](https://github.com/apple/swift-nio) for its networking layer.
 
 ### Commits and pull requests
 
 > TODO
 
 ### Accepted proposals
 
+[SE-0235](https://github.com/apple/swift-evolution/blob/master/proposals/0235-add-result.md): *Add Result to the Standard Library* was [accepted]().
```
Comment (BasThomas) [inline _drafts/2018-12-13-issue-124.md]: This is still incomplete.

File: _drafts/2018-12-13-issue-124.md [hunk 2]
```diff
@@ -21,27 +25,49 @@ sponsor:
...
+
 [SE-0236](https://github.com/apple/swift-evolution/blob/master/proposals/0236-package-manager-platform-deployment-settings.md): *Package Manager Platform Deployment Settings* was [accepted](https://forums.swift.org/t/accepted-with-modifications-se-0236-package-manager-platform-deployment-settings/18420).
 
 > There was a lot of negative feedback about the platform restrictions part of the proposal, so the SwiftPM code owners decided to accept a revised version of the proposal which only handles customization of deployment targets. The topic of platform restrictions for packages can be revisited separately at a later time.
 >
 > For specifying custom deployment versions, the original proposal mentions that an initializer will be provided. However, since that would be too verbose, a string overload will be provided instead. This makes the API `.macOS("10.13")` instead of `.macOS(MacOSVersion("10.13"))`.
 
+[SE-0238](https://github.com/apple/swift-evolution/blob/master/proposals/0238-package-manager-build-settings.md): *Package Manager Target Specific Build Settings* was [accepted]().
```
Comment (BasThomas) [inline _drafts/2018-12-13-issue-124.md]: And this.

File: _drafts/2018-12-13-issue-124.md [hunk 3]
```diff
@@ -21,27 +25,49 @@ sponsor:
...
+
+> This is a proposal for adding support for declaring some commonly used target-specific build settings in the Package.swift manifest file. As the name suggests, target-specific build settings are only applied to a particular target. SwiftPM also aims to support cross-target build settings that go across the target boundary and impart certain settings on a target's dependees, but this proposal is only concerned with the former type of build settings and the latter will be explored with a future proposal.
+
 ### Returned proposals
 
-> TODO
+Review #2 of [SE-0237](https://github.com/apple/swift-evolution/blob/master/proposals/0237-contiguous-collection.md): *Introduce `withUnsafe{Mutable}BufferPointerIfSupported` methods* was [returned for revision]().
```
Comment (BasThomas) [inline _drafts/2018-12-13-issue-124.md]: So is this one.

File: _drafts/2018-12-13-issue-124.md [hunk 4]
```diff
@@ -9,7 +9,11 @@ sponsor:
     displaylink: TODO
 ---
 
-> TODO: intro comments
```
Comment (BasThomas) [inline _drafts/2018-12-13-issue-124.md]: Feel free to introduce yourself as a writer! :)

File: _drafts/2018-12-13-issue-124.md [hunk 5]
```diff
@@ -9,39 +9,88 @@ sponsor:
...
+
+[Paul Hudson](https://twitter.com/twostraws) wrote [an article](https://www.hackingwithswift.com/articles/134/how-to-use-dynamiccallable-in-swift) explaining how to use `@dynamicCallable` in Swift.
+
+[Bruno Rocha](https://twitter.com/rockthebruno) explored [how @dynamicMemberLookup Works Internally in Swift (+ Creating Custom Swift Attributes)](https://swiftrocks.com/how-dynamicmemberlookup-works-internally-in-swift.html)
+
+[Ole Begemann](https://twitter.com/olebegemann) investigated how to [split a Swift Sequence into head and tail](https://oleb.net/2018/sequence-head-tail/) in Swift 4.2 and why it will break in Swift 5 because of [SE-0234](https://github.com/apple/swift-evolution/blob/master/proposals/0234-remove-sequence-subsequence.md).
+
+Amazon has open sourced [`smoke-framework`](https://github.com/amzn/smoke-framework), a lightweight server-side service framework written in Swift and using [SwiftNIO](https://github.com/apple/swift-nio) for its networking layer.
+
+Videos from [Functional Swift Conference](http://2018.funswiftconf.com/) that happened December 1st [have been released](https://twitter.com/mbrandonw/status/1072541950954352640).
```
Comment (BasThomas) [inline _drafts/2018-12-13-issue-124.md]: Although I think this was an awesome conference, I am wondering how related this is to the Swift open source projects. Are there any particular talks that discuss Swift open source projects?

Comment (fassko) [reply _drafts/2018-12-13-issue-124.md]: I haven't watched all but this spotted my eye https://www.youtube.com/watch?v=BXJIIQ-B4-E&feature=youtu.be

File: _drafts/2018-12-13-issue-124.md [hunk 6]
```diff
@@ -2,14 +2,9 @@
 layout: post
 title: ! 'Issue #124'
 author: fassko
-sponsor:
```
Comment (BasThomas) [inline _drafts/2018-12-13-issue-124.md]: Removed this as it is not needed.

File: _drafts/2018-12-13-issue-124.md [hunk 7]
```diff
@@ -21,45 +16,40 @@ A lot has happened in the Swift community and it seems that team wants to finish
...
-Jesse and JP [explored](https://spec.fm/podcasts/swift-unwrapped/234517) the famous [Result proposal](https://github.com/apple/swift-evolution/blob/master/proposals/0235-add-result.md), going over its pros and cons.
+Jesse and JP [explored](https://spec.fm/podcasts/swift-unwrapped/234517) the famous [Result proposal](https://github.com/apple/swift-evolution/blob/master/proposals/0235-add-result.md), discussing its pros and cons.
 
 ### News and community
 
 The Swift Server Work Group had a meeting. Not only did they add new members, but they also [discussed some important topics](https://forums.swift.org/t/november-29th-2018/18400) such as Docker updates, Crypto, HTTP Client improvements and more.
 
-[Yasuhiro Inami](https://twitter.com/inamiy) is announcing [SwiftRewriter](https://github.com/inamiy/SwiftRewriter), a composable Swift code formatter on top of [SwiftSyntax](https://github.com/apple/swift-syntax).
-
-[Paul Hudson](https://twitter.com/twostraws) wrote [an article](https://www.hackingwithswift.com/articles/134/how-to-use-dynamiccallable-in-swift) explaining how to use `@dynamicCallable` in Swift.
```
Comment (BasThomas) [inline _drafts/2018-12-13-issue-124.md]: Removed this as it didn't feel relevant enough, and we have a bunch of news already :)

File: _drafts/2018-12-13-issue-124.md [hunk 8]
```diff
@@ -21,45 +16,40 @@ A lot has happened in the Swift community and it seems that team wants to finish
...
+[Slava Pestov](https://github.com/slavapestov) merged [a pull request](https://github.com/apple/swift/pull/21155) that fixes converting a metatype to a readable string when the metatype is for a function type with a single tuple type argument. Related to [SR-8235](https://bugs.swift.org/browse/SR-8235).
 
 [Joe Groff](https://github.com/jckarter) merged [a pull request](https://github.com/apple/swift/pull/21102) that provides ABI space for source location info in unconditional casts.
 
-[Harlan Haskins](https://github.com/harlanhaskins) merged [a pull request](https://github.com/apple/swift/pull/21033) that consolidates the functions used to check accessors vs. other decls, and makes sure we check setter access as well as regular decl access. This also resolves [SR-8969](https://bugs.swift.org/browse/SR-8969).
-
-[Mark Lacey](https://github.com/rudkx) opened [a pull request](https://github.com/apple/swift-evolution/pull/966) creating proposal SE-0239 *Extending operator declarations with designated type*.
```
Comment (BasThomas) [inline _drafts/2018-12-13-issue-124.md]: Removed this one as it is not merged yet.

File: _drafts/2018-12-13-issue-124.md [hunk 9]
```diff
@@ -126,13 +110,13 @@ We all know [Mattt](https://twitter.com/mattt)'s [obsession of Strings](https://
...
+A discussion has arisen about [KeyPaths for Tuples in Swift 5](https://forums.swift.org/t/keypaths-tuples-and-swift-5/18465).
 
 > .. finding out that `\SomeType.aTuple.0` does not work ..
 >
 > I think this would be a great starter project. The runtime ABI for keypaths should not need any modification to handle tuple components, so there's no rush to implement this. There's an [ABI document](https://github.com/apple/swift/blob/master/docs/ABI/KeyPaths.md) that describes the layout of key path objects; the details are somewhat out of date, but the high level structure is the same
 
-Very interesting discussion about [will Diagnostics and Type Inference Get Some Love after ABI stability](https://forums.swift.org/t/after-abi-stability-will-diagnostics-and-type-inference-get-some-love/18685).
+Very interesting discussion about [Diagnostics and Type Inference after ABI stability](https://forums.swift.org/t/after-abi-stability-will-diagnostics-and-type-inference-get-some-love/18685).
 
 > Two things I really wish would improve are diagnostics and type inference, in particular when it comes to closures and generics. (I'm beginning to accept the limitations on PATs).
```
Comment (BasThomas) [inline _drafts/2018-12-13-issue-124.md]: Do you remember / know what a `PAT` is again? If so, can we use the full name?

Comment (fassko) [reply _drafts/2018-12-13-issue-124.md]: Will change to Protocol with Associated Types as not all folks know this acronym.

Comment (BasThomas) [reply _drafts/2018-12-13-issue-124.md]: Ahh, it was on the tip of my tongue. Nice :)

**Comment by BasThomas** [comment]
Thanks so much @fassko! 🎉

**Comment by jessesquires** [comment]
Hey @fassko ! Great job on your first issue! Thanks so much for contributing! 💯 🙌

**Comment by fassko** [comment]
Thanks for trusting me and giving this chance. 😊

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**